repo.go

```diff
@@ -167,6 +167,45 @@ func (r *Repo) RootVersion() (int, error) {
 	return root.Version, nil
 }
 
+func (r *Repo) GetThreshold(keyRole string) (int, error) {
+	root, err := r.root()
+	if err != nil {
+		return -1, err
+	}
+	role, ok := root.Roles[keyRole]
+	if !ok {
+		return -1, ErrInvalidRole{keyRole}
+	}
+
+	return role.Threshold, nil
+}
+
+func (r *Repo) SetThreshold(keyRole string, t int) error {
+	if !validManifest(keyRole + ".json") {
+		// Delegations are not currently supported, so return an error if this is not a
+		// top-level manifest.
+		return ErrInvalidRole{keyRole}
+	}
+	root, err := r.root()
+	if err != nil {
+		return err
+	}
+	role, ok := root.Roles[keyRole]
+	if !ok {
+		return ErrInvalidRole{keyRole}
+	}
+	if role.Threshold == t {
+		// Change was a no-op.
+		return nil
+	}
+	role.Threshold = t
+	if _, ok := r.versionUpdated["root.json"]; !ok {
+		root.Version++
+		r.versionUpdated["root.json"] = struct{}{}
```

Comment:
Interesting: what does this line do?

Reply:
Ah so `versionUpdated` is just a map that stores whether the role has been version incremented or not before committing. So if it hasn't yet been incremented (because `r.versionUpdated["root.json"]` entry isn't there), only then will be update and then store the entry in the map. This prevents incrementing versions multiple changes before committing.

Tbh I don't know why this is a map into an empty struct, rather than a boolean..